course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/ContentConsumptionActor.scala

```diff
@@ -114,7 +114,12 @@ class ContentConsumptionActor @Inject() extends BaseEnrolmentActor {
                             if(entry._2.head.containsKey(JsonKey.COLLECTION_ID)) entry._2.head.remove(JsonKey.COLLECTION_ID)
                             val contentIds = entry._2.map(e => e.getOrDefault(JsonKey.CONTENT_ID, "").asInstanceOf[String]).asJava
                             val existingContents = getContentsConsumption(userId, courseId, contentIds, batchId, request.getRequestContext).groupBy(x => x.get("contentId").asInstanceOf[String]).map(e => e._1 -> e._2.toList.head).toMap
+                            //added progress....
+                            val cprogress = getProgress(entry._2)
+                            cassandraOperation.batchInsert(request.getRequestContext, consumptionDBInfo.getKeySpace, "user_content_consumption_progress", cprogress)
```

Comment:
@pritha-tarento : what is the rationale for using a new table instead of adding a new column "progressDetails" to the existing table?

Reply:
Current table read/fetch was not supporting the blob content. Hence the original idea to support it separately helps to maintain easy. Also including the fact for not to break the existing functionality.